src/lib/dialog_ui.sh

```diff
@@ -433,17 +437,30 @@ function create_choice_list_screen()
   cmd+=" --radiolist $'${message_box}'"
   # Set height and width
   cmd+=" '${height}' '${width}' '0'"
-  # Add choices (${choices["${choice}"]} refers to the choice's description)
-  for choice in "${!_choices[@]}"; do
-    choice_description=$(str_escape_single_quotes "${_choices["${choice}"]}")
-    choice=$(str_escape_single_quotes "$choice")
-    if [[ "${_check_statuses["$index"]}" == 1 ]]; then
-      cmd+=" $'${choice}' $'${choice_description}' 'on'"
-    else
-      cmd+=" $'${choice}' $'${choice_description}' 'off'"
-    fi
-    ((index++))
-  done
+  # Add choices without description if `_choices` is an indexed array
+  if [[ "$(typeset -p "$3")" =~ 'declare -a' ]]; then
+    for choice in "${_choices[@]}"; do
+      choice=$(str_escape_single_quotes "$choice")
+      if [[ "${_check_statuses["$index"]}" == 1 ]]; then
```

Comment:
I believe that the reason that the default option was not working correctly came from the fact that you mapped one array (`_check_statuses`) against an associative array (`_choices`), and associative arrays do not follow the same order as a regular array. Before your change, you were trying to use an array (index) to match an associative array order that can not be guaranteed. If you change `_check_status` to an associative array, and require the user to use the same key for `_check_statuses` and `_choices`, you can map the default options.

Reply:
Sorry for the messy commit message, but the issue wasn't that `_choices` and `_check_statuses` were not in sync, as I make the caller populate `_check_statuses` in the same manner that we do the matching in this function, i.e., with a foreach loop which has a consistent order when looping through keys of an associative array across multiple "calls". The problem was that this order of keys isn't the same as the order of declaration, neither the lexicographic order, like you can see in this screenshot:

<img width="1059" alt="Screen Shot 2023-08-20 at 18 34 34" src="https://github.com/kworkflow/kworkflow/assets/95055784/62060649-acaa-4404-92e3-0e6e564c2461">

This made it virtually impossible for the caller to define an order of displaying the choices with an associative array, which was the problem that I tried tackling with this commit.